test/evergreen.yml

```diff
@@ -71,59 +71,64 @@ functions:
             install_name_tool -change /usr/local/lib/libwiredtiger-$WT_VERSION.dylib $(pwd)/.libs/libwiredtiger-$WT_VERSION.dylib .libs/wt
           fi
         fi
-
-pre:
-  - command: shell.exec
-    params:
-      script: |
-        rm -rf "wiredtiger"
-post:
-  - command: shell.exec
+  "make check directory":
+    command: shell.exec
     params:
-      working_dir: "wiredtiger"
+      working_dir: "wiredtiger/build_posix"
       script: |
         set -o errexit
         set -o verbose
-        tar cfz ../wiredtiger.tgz .
-  - command: s3.put
+
+        ${test_env_vars|} ${make_command|make} VERBOSE=1 check -C ${directory}  ${smp_command|} 2>&1
+  "upload artifact":
+    - command: archive.targz_pack
+      params:
+        target: "wiredtiger.tgz"
+        source_dir: "wiredtiger"
+        include:
+          - "./**"
+    - command: s3.put
+      params:
+        aws_secret: ${aws_secret}
+        aws_key: ${aws_key}
+        local_file: wiredtiger.tgz
+        bucket: build_external
+        permissions: public-read
+        content_type: application/tar
+        display_name: Artifacts
+        remote_file: wiredtiger/${build_variant}/${revision}/artifacts/${task_name}_${build_id}.tgz
+  "cleanup":
+    command: shell.exec
     params:
-      aws_secret: ${aws_secret}
-      aws_key: ${aws_key}
-      local_file: wiredtiger.tgz
-      bucket: build_external
-      permissions: public-read
-      content_type: application/tar
-      display_name: Artifacts
-      remote_file: wiredtiger/${build_variant}/${revision}/artifacts/${task_id}.tgz
+      script: |
+        rm -rf "wiredtiger"
+        rm -rf "wiredtiger.tgz"
+
+pre:
+  - func: "cleanup"
+post:
+  - func: "upload artifact"
+  - func: "cleanup"
```

Comment:
This may not be necessary as Evergreen usually takes care of the cleanup.

Reply:
Better to keep that here. In case some of the stage failed to run.